src/adaptors/UBSvgSubsetAdaptor.cpp

```diff
@@ -2606,6 +2914,271 @@ void UBSvgSubsetAdaptor::UBSvgSubsetWriter::textItemToSvg(UBGraphicsTextItem* it
+              }
+              // Note: do not use font when baseline-shift is given
+              // as this makes the text too small.  In SMART output the
+              // original font size was given in the supersuboriginalfont
+              // attribute.
+              curCursor.mergeCharFormat(format);
+              textItem->setTextCursor(curCursor);
+
+              // just the following doesn't seem to do anything
+              //textItem->setFont(font);
+
+              last_y = y;
+
+              // Turn off text wrapping so that the width of the text is
+              // automatically increased.
+              QTextOption option = textItem->document()->defaultTextOption();
+              option.setWrapMode(QTextOption::NoWrap);
+              textItem->document()->setDefaultTextOption(option);
+
+              curCursor.insertText(text);
+
+              // Attempt to make the text item wide enough, and turn
+              // word wrap back on.
+              // I don't know why it's necessary but increase the width
+              // by 100, otherwise the text is slightly too narrow after
+              // saving and reloading.
+              QSizeF size = textItem->document()->size();
+              textItem->setTextWidth(size.width() + 100);
+              option.setWrapMode(QTextOption::WordWrap);
+              textItem->document()->setDefaultTextOption(option);
+          }
+        }
+        else if (mXmlReader.isStartElement() && mXmlReader.name() == "tspan")
+        {
+          extractSvgText (font, dx, dy, textEnd, textItem, last_y);
+        }
+        tt = mXmlReader.readNext();
+    }
+}
+
+void UBSvgSubsetAdaptor::UBSvgSubsetReader::addTextItemsFromSvg()
+{
+    qreal dx = 0.0;
+    qreal dy = 0.0;
+    QXmlStreamAttributes a = mXmlReader.attributes();
+    if (!a.value("transform").isNull())
+    {
+        QString trString = a.value("transform").toString();
+        doTransform(trString, dx, dy);
+    }
+
+    qreal fontSize = 12;
+    QColor fontColor(qApp->palette().foreground().color());
+    QString fontFamily = "Arial";
+    QString fontStretch = "normal";
+    bool italic = false;
+    int fontWeight = QFont::Normal;
+    parseTextAttributes(fontSize, fontColor, fontFamily, fontStretch, italic, fontWeight);
+    QFont startFont(fontFamily, fontSize, fontWeight, italic);
+
+    qreal textEnd = 0;
+    UBGraphicsTextItem *textItem = 0;
+    qreal last_y = 0;
+    extractSvgText(startFont, dx, dy, textEnd, textItem, last_y);
+
+    return;
+}
+
+
+// extract dx and dy from SVG transform attribute.  other transformations
+// (like rotations) are ignored.
+void UBSvgSubsetAdaptor::UBSvgSubsetReader::doTransform (QString& trString, qreal& dx, qreal& dy)
+{
+#if 0
+  qreal dxr = 0.0;
+  qreal dyr = 0.0;
+  qreal dx = 0.0;
+  qreal dy = 0.0;
+  qreal angle = 0.0;
+#endif
+
+  foreach(QString trStr, trString.split(" ", QString::SkipEmptyParts))
+  {
+      QRegExp regexp;
+#if 0
```

Comment:
dead code

Reply:
It's there in case those parts of the transformation will be supported in the future. I'll remove the code if the maintainers ask me to.

Reply:
I should add I copied and pasted this code from the CFF Import code (which doesn't work, as far as I can tell).